### ISSUE TITLE
[bug] Inability to execute write calls to contracts on React Native

### ISSUE THREAD

**Issue by natedevxyz** [WalletConnect/web3modal-react-native#116]
### Expected Behavior
The useContractWrite function from wagmi should successfully enable write calls to contracts when implemented within React Native, following the instructions provided in the official WalletConnect documentation.

### Current Behavior
The useContractWrite function executes write calls to contracts; however, it throws an error resulting in a disruption of the application's functionality.

![](https://cdn.discordapp.com/attachments/1105316298390511626/1184581115160297552/Screenshot_2023-12-13_at_16.37.55.png?ex=658c7e22&is=657a0922&hm=9d870604c1e900528de45f7b51c24e9104224f066308b711ba81580a1e9ae226&)

### Steps to Reproduce
1. Set up a React Native project.
2. Install all required packages and dependencies as per the official documentation.
3. Perform the initial setup of Web3Modal.
4. Import and utilize the useContractWrite function from wagmi.
5. Attempt to execute a write call to a contract.

### Context (Environment)
The issue occurs within the following environment:

- "@web3modal/wagmi-react-native": "^1.0.1"
- "expo": "~49.0.15"
- "react-native": "0.72.6"
- "viem": "^1.20.3"
- "wagmi": "^1.4.12"

This issue impacts all developers seeking to integrate the useContractWrite function into React Native applications using Web3Modal.

### Possible Solution
The problem seems to be a function inside @walletconnect/sign-client/node_modules/@walletconnect/utils/dist/index.cjs.js that uses localStorage. The solution could be to use AsyncStorage instead.

https://react-native-async-storage.github.io/async-storage/

### Detailed Description
To reproduce and examine the issue firsthand within an Expo environment, please refer to the repository linked below:

https://github.com/natedevxyz/web3modal-rn-test

**Comment by ignaciosantise** [comment]
Hey @natedevxyz 👋 thanks for the report (super clear btw)

I was able to reproduce the bug, and the fix being reviewed.

**Comment by natedevxyz** [comment]
Hey @ignaciosantise, I saw that a fix was merged to the walletconnect monorepo. Do you have an idea when it might be published?

**Comment by ignaciosantise** [comment]
hey @natedevxyz 👋 this week we'll create a PR on wagmi to update their walletconnect dependencies and then we'll have to wait until they release a new version. As a workaround until then, you can override the version in your package.json

you can do something like this:

```json
//package.json

"dependencies": {
   ...
},
"resolutions": {
   "@walletconnect/utils": "2.11.0",
   "@walletconnect/ethereum-provider": "2.11.0"
}
```

**Comment by natedevxyz** [comment]
> hey @natedevxyz 👋 we created a [PR](https://github.com/wevm/wagmi/pull/3361) on wagmi to update their walletconnect dependencies and then we'll have to wait until they release a new version. As a workaround until then, you can override the version in your package.json
> 
> you can do something like this:
> 
> ```json
> //package.json
> 
> "dependencies": {
>    ...
> },
> "resolutions": {
>    "@walletconnect/utils": "2.11.0",
>    "@walletconnect/ethereum-provider": "2.11.0"
> }
> ```

I tried this but I keep getting the same error. I deleted package-lock.json and the node_modules folder after adding the resolutions field.

**Comment by ignaciosantise** [comment]
If you are using npm, replace resolutions with "overrides" and reinstall everything again

**Comment by natedevxyz** [comment]
> If you are using npm, replace resolutions with "overrides" and reinstall everything again

Thank you, that did it :raised_hands:

**Comment by ignaciosantise** [comment]
@natedevxyz wagmi released the fix in`v1.4.13`

Thanks for reporting 🎉
